service.js

```diff
@@ -1,42 +1,20 @@
-const service = require('service-systemd')
-const pkg = require('./package.json')
-const path = require('path')
+const PORT = parseInt(process.env.PORT || '3000')
+const OSC_PORT = parseInt(process.env.OSC_PORT || '41234')
 
-function add () {
-  service
-    .add({
-      name: pkg.name,
-      cwd: '.',
-      app: path.resolve(`./${pkg.main}`),
-      user: 'room',
-      engine: 'node',
-      pid: `/var/run/${pkg.name}.pid`,
-      log: `/var/log/${pkg.name}/log`,
-      error: `/var/log/${pkg.name}/error`,
-      'engine.bin': process.env.NODE,
-      env: {
-        NODE_ENV: 'production'
-      }
-    })
-    .then(() => console.log(`${pkg.name} service removed`))
-    .catch(err => console.error(`error removing ${pkg.name}`, err.toString()))
-}
+const Database = require('@living-room/database-js')
+const room = new Database()
 
-function remove () {
-  service
-    .remove(pkg.name)
-    .then(() => console.log(`${pkg.name} service removed`))
-    .catch(err => console.error(`error removing ${pkg.name}`, err.toString()))
-}
+const ServiceManager = require('./src/ServiceManager')
+const serviceManager = new ServiceManager()
 
-switch (process.argv[2]) {
-  case 'add':
-    add()
-    break
-  case 'remove':
-    remove()
-    break
-  default:
-    remove()
-    add()
-}
+let app = require('./lib/httpServer.js')(room.client('http'))
```

Comment:
This currying stuff ends up being kind of hard to read/follow in the next couple of lines.

Reply:
Yeah, i'm going to refactor into something like `const httpService = require(`./src/HttpService`).create(room.client('http'), { options })`